regulations/urls.py

```diff
@@ -27,156 +27,153 @@
+        lt_cache(ChromeSectionDiffView.as_view()), name='chrome_section_diff_view'),
     url(r'^preamble/(?P<doc_number>[\w-]+)/cfr_changes/(?P<section>[\w-]+)$',
         daily_cache(CFRChangesView.as_view()), name='cfr_changes'),
     url(r'^preamble/(?P<paragraphs>[-\w]+(/[-\w]+)*)$',
         daily_cache(PreambleView.as_view()), name='chrome_preamble'),
 
     # Redirect to version by date
     # Example: http://.../201-3-v/1999/11/8
-    url(r'^%s/(?P<year>\d{4})/(?P<month>\d{2})/(?P<day>\d{2})$'
-        % paragraph_pattern, redirect_by_date, name='redirect_by_date'),
+    url(r'^(?P<label_id>[-\w]+)/(?P<year>\d{4})/(?P<month>\d{2})/(?P<day>\d{2})$',
+        redirect_by_date, name='redirect_by_date'),
+
     # Redirect to version by current date
     # Example: http://.../201-3-v/CURRENT
-    url(r'^%s/CURRENT$' % paragraph_pattern,
+    url(r'^(?P<label_id>[-\w]+)/CURRENT$',
         redirect_by_current_date, name='redirect_by_current_date'),
+
     # A regulation section with chrome
     # Example: http://.../201-4/2013-10704
-    url(r'^%s/%s$' % (section_pattern, version_pattern),
+    url(r'^(?P<label_id>[\d]+[-][\w]+)/(?P<version>[-\d\w_]+)$',
         lt_cache(ChromeView.as_view(partial_class=PartialSectionView)),
         name='chrome_section_view'),
+
     # Subterp, interpretations of a while subpart, emptypart or appendices
     # Example: http://.../201-Subpart-A-Interp/2013-10706
     #          http://.../201-Subpart-Interp/2013-10706
     #          http://.../201-Appendices-Interp/2013-10706
-    url(r'^%s/%s$' % (subterp_pattern, version_pattern),
-        lt_cache(ChromeSubterpView.as_view()),
-        name=ChromeSubterpView.version_switch_view),
+    url(r'^(?P<label_id>[\d]+-(Appendices|Subpart(-[A-Z]+)?)-Interp)/(?P<version>[-\d\w_]+)$',
+        lt_cache(ChromeSubterpView.as_view()), name=ChromeSubterpView.version_switch_view),
+
     # Interpretation of a section/paragraph or appendix
     # Example: http://.../201-4-Interp/2013-10704
-    url(r'^%s/%s$' % (interp_pattern, version_pattern),
-        lt_cache(ChromeView.as_view(
-            partial_class=partial_interp.PartialInterpView)),
+    url(r'^(?P<label_id>[-\w]+[-]Interp)/(?P<version>[-\d\w_]+)$',
+        lt_cache(ChromeView.as_view(partial_class=partial_interp.PartialInterpView)),
         name='chrome_interp_view'),
+
     # The whole regulation with chrome
     # Example: http://.../201/2013-10704
-    url(r'^%s/%s$' % (reg_pattern, version_pattern),
+    url(r'^(?P<label_id>[\d]+)/(?P<version>[-\d\w_]+)$',
         lt_cache(ChromeView.as_view(
             partial_class=PartialRegulationView,
             version_switch_view='chrome_regulation_view')),
         name='chrome_regulation_view'),
+
     # A regulation paragraph with chrome
     # Example: http://.../201-2-g/2013-10704
-    url(r'^%s/%s$' % (paragraph_pattern, version_pattern),
+    url(r'^(?P<label_id>[-\w]+)/(?P<version>[-\d\w_]+)$',
```

Comment:
Kind of related to the above. If there's a way to emphasize the desired match clearly, this pattern should be more concrete.